### ISSUE TITLE
Improve warning message when VS Code extension detects Node.js 22 incompatible local markuplint (v4.0.0-4.9.x)

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#2838]
## Summary

**Enhancement proposal**: When the VS Code extension detects import assertion syntax errors from local markuplint v4.0.0-4.9.x, it should display a more user-friendly warning message explaining the compatibility issue and providing clear guidance.

## Current Behavior
When loading local markuplint v4.0.0-4.9.x with Node.js 22+ (VS Code 1.90.0+):
- Extension logs: "Local markuplint package has import assertion compatibility issues"
- Extension silently falls back to bundled version
- User may not understand why local markuplint isn't being used

## Proposed Enhancement
Display a clear, actionable warning message when `assert { type: 'json' }` syntax error is detected:

### Suggested Warning Message:
```
⚠️ Local markuplint compatibility issue detected

Your local markuplint version (v4.X.X) is incompatible with VS Code's Node.js 22 engine.
Using bundled markuplint version instead.

To use your local markuplint version:
• Upgrade to markuplint@4.10.0 or later
• Or downgrade VS Code to version < 1.90.0

See: https://github.com/markuplint/markuplint/issues/2837
```

## Implementation Details
- **Trigger**: When `SyntaxError: Unexpected identifier 'assert'` is caught during local markuplint import
- **Display method**: VS Code notification popup + output channel warning
- **Frequency**: Once per session to avoid spam
- **Additional**: Include link to documentation/issue for more details

## Benefits
- ✅ **User awareness**: Clear explanation of why local markuplint isn't working
- ✅ **Actionable guidance**: Specific steps to resolve the issue
- ✅ **Better UX**: Users understand the fallback behavior
- ✅ **Reduced confusion**: No more "why isn't my local version being used?"

## Technical Notes
- Should be implemented in `src/server/get-module.ts` where the error is currently caught
- Can reuse existing LSP notification infrastructure
- Should include version detection to show specific affected version number

## Related
- #2837 - Documentation of the underlying compatibility issue

This enhancement will significantly improve the user experience when encountering this known compatibility issue.
